numpy/compat/py3k.py

```diff
@@ -1,9 +1,15 @@
 """
-Python 3 compatibility tools.
+Python 3.X compatibility tools.
 
-"""
-from __future__ import division, absolute_import, print_function
+While this file was originally intented for Python 2 -> 3 transition,
+it is now used to create a compatibility layer between different
+minor versions of Python 3.
 
+While the active version of numpy may not support a given version of python, we
+allow downstream libraries to continue to use these shims for forward
+compatibility with numpy while they transition their code to newer versions of
+Python.
+"""
```

Comment:
Great comment, thanks - to the extent that I'd consider merging a PR with only this change in for 1.18.0, if it's worth your time to rebase.